src/main/java/recipe/RecipeReadWriter.java

```diff
@@ -3,19 +3,18 @@
 import entities.RecipeList;
 
 import java.io.*;
-import java.util.ArrayList;
 
-public class RecipeRepoImpl implements RecipeRepoGateway{
+public class RecipeReadWriter implements RecipeRepoGateway{
     private static final String file = "recipe.sav";
-    private static RecipeRepoImpl recipeRepoImpl;
+    private static RecipeReadWriter recipeReadWriter;
 
-    private RecipeRepoImpl(){}
+    private RecipeReadWriter(){}
 
-    public static RecipeRepoImpl getRecipeRepoImpl(){
-        if (recipeRepoImpl == null){
-            recipeRepoImpl = new RecipeRepoImpl();
+    public static RecipeReadWriter getRecipeRepo(){
```

Comment:
In general, it may help to add some javadoc comments for clearer documentation on what each method does. I believe it's a requirement for the rubric as well